### PR TITLE
PR for completed Outputs

### PR DESCRIPTION
**Pull request by tezebuike**
**What does this PR do?**

This PR shows the output for the following:
1 Setting up continuous integration
2 Writing frontend with html-js-css.
3 Checking-repository-health
4 Solving the problem
5 Writing test
6 Delivering your product
7 Setting up software development environments(deployment)
8 Reflecting-on-the-learning-experience
9 Automating tasks
10 Getting peer reviews
11 setting-up-linters
12 using-design-patterns
13 setting-up-your-repository

**Description of Task to be completed?**

1 A link to a successful TravisCI build of your development branch 
2 A folder containing the template for the proposed front-end of my application
3 A screenshot of code coverage and GPA badges in my repo's README file
4 The file containing the implementation.
5 The test files for the Inverted Index class.
6 Google Doc Release notes for your final product in your repo's wiki
7 Link to live URLs to a review app, staging & production apps
8 A post reflecting on your learning experience while producing these outputs
9 A link to a video showing how gulp automates tasks
10 A Pull Request showing at least two peer code reviews
11 Two screenshots of ESLint Airbnb style violations in the terminal and HoundCI comment on a PR
12 A detailed explanation of Design Patterns used while solving-the-problem
13 A link to the repository for my project with at least one Pull Request

How should this be manually tested?
N/A

### CODE REVIEW

File: 06-building-a-program-start-to-finish/automating-tasks/Output/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Link to YouTube video
+- [Gulp Screencast](https://youtu.be/cm9B84_zT0M)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/automating-tasks/Output/README.md:7]: Looks good!

File: 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+#### Screenshots of Code coverage and GPA badges
+![Badges](https://github.com/andela/test-simulations-mgmt-repo/blob/bayo-kakashi-tezebuike/06-building-a-program-start-to-finish/setting-up-linters/output/coverage.png)
+
+
+#### Targeted Learning Outcomes
+- [End-to-End Testing](https://github.com/andela/learningmap/tree/master/Phase-C/Entry-level%20Developer/Curriculum/37%20-%20End-to-End%20Testing)
+- [Test-Driven Development](https://github.com/andela/learningmap/tree/master/Phase-C/Entry-level%20Developer/Curriculum/25%20-%20Test-Driven%20Development)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md:12]: Seen!

File: 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+####Link to Release Notes
+Can be found [here](https://github.com/andela-tezebuike/inverted-index/wiki/Release-Notes)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md:7]: What you have here is basically a copy of your `ReadMe`, please give information relevant to the `Release` of your product.

File: 06-building-a-program-start-to-finish/getting-peer-reviews/Output/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: GitHub
+- **Deliver**: a Pull Request showing at least two Peer Code Reviews
+- **With**: *independently*
+
+#### Link to Peer Review
+- [Peer Review](https://github.com/andela-tezebuike/inverted-index/pull/4)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/getting-peer-reviews/Output/README.md:7]: Seen!

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#### Output
+- **Using**: Medium
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+####Here is the link to the medium post:
+- [Nuggets on the journey] (https://medium.com/@Tracezeb/nuggets-on-the-journey-bcd36e77cbeb#.l40tu2xva)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md:7]: This was a good read.

It however said nothing about "your learning experience while producing these outputs". Nevertheless, a good one.

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#### Output
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.md:1]: I don't see the link to your TravisCI build here.

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md [hunk 7]
```diff
@@ -3,6 +3,12 @@
 - **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
 - **With**: *independently*
 
+### Link to a successful Travis build on development branch
+[Travis development branch - Andela-Judensi/inverted-index](https://travis-ci.org/andela-tezebuike/inverted-index/jobs/217681187)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md]: Why is this here: "Travis development branch - Andela-Judensi/inverted-index"?

Please let's have just one file for this output.

Also, include a verbose reporter for your tests. That way we can see the details of the execution.

File: 06-building-a-program-start-to-finish/setting-up-linters/Output/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+#### Output
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-linters/Output/README.md:1]: Seen!

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+#### Output
+- **Using**: Heroku
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Here are links to the `review`, `staging` & `production` apps
+ - [Review]
+ (https://invertedindex-review.herokuapp.com/index.html)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md]: I expect that this is a review that's generated from your PRs automatically on Heroku. The URL isn't consistent with what that action performs.

Please review this.

File: 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+
+#### Output
+- **Using**: GitHub
+- **Deliver**: the repository for your proposed project and at least one `Pull Request`
+- **With**: *independently*
+
+#### Click here to go to the Github repo
+- [andela-tezebuike/inverted-index]
+(https://github.com/andela-tezebuike/inverted-index.git)
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md]: Please include details on how people can contribute to this project in the ReadMe.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 11]
```diff
@@ -0,0 +1,142 @@
+/**
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Please ensure this file's name is in PascalCase as prescribed by our style guides.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+   * Create index
+   * @function
+   * @param {string} fileName
+   * @param {Array} books
+   * @return {Object} index object
+   */
+  createIndex(fileName, books) {
+    const indices = {};
+    books.forEach((book, index) => {
+      let words = '';
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: I believe it's alright to join the declaration of this variable with the first use in line 99.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+    return this.indexed[fileName];
+  }
+
+  /**
+   * Read files using FileReader
+   * @function
+   * @param {String} file
+   * @return {Promise} validateFile response
+   */
+  static readFile(file) {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: This right here, is beautiful work!

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    });
+  }
+
+  /**
+   * Validate File
+   * @function
+   * @param {string} fileToValidate content of the file uploaded
+   * @return {Object} result.success should return true or false
+   */
+  static validateFile(fileToValidate) {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Nice work here!

I see you're looping through all the files to validate each of the files. I however can see a logical error here, if you have three documents in a file and the first two are invalid with the last file valid it will still return true for that file.

Is the end goal to test that at least one document is valid or all documents in a file are valid?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+   * @param {String} words text to be tokenized.
+   * @return {Array} array of string tokens
+   */
+  static tokenize(words) {
+    const pattern = /[ .:;?!~,`'&|()<>{}[\]\r\n/\\]+/;
+    return words.toLowerCase().split(pattern);
+  }
+
+  /**
+   * Create index
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Your JSDoc barely communicates anything here. Please review.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+   * @function
+   * @param {string} fileName
+   * @param {Array} books
+   * @return {Object} index object
+   */
+  createIndex(fileName, books) {
+    const indices = {};
+    books.forEach((book, index) => {
+      let words = '';
+      words = (`${book.title} ${book.text}`);
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: You actually don't need the extra parenthesis.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+      words = InvertedIndex.tokenize(words);
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: I believe there are better ways to name variables rather than reuse `words` for multiple assignments of entirely different values and datatypes.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+    };
+  }
+
+  /**
+   * Search Index.
+   * @function
+   * @param {String} phrase search string
+   * @returns {Object} search result object.
+   */
+  searchIndex(phrase) {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Improve JSDoc.

Also, can I search through a particular file and not all uploaded files at once?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+      searchWords.forEach((word) => {
+        if (storedIndex.eachWord[word]) {
+          search.eachWord[word] = storedIndex.eachWord[word];
+        } else search.eachWord[word] = [];
+      });
+      result[filename] = search;
+    });
+    return result;
+  }
+}
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: I don't see you ensuring that the words in an index are unique. Is that catered for somewhere?

File: 06-building-a-program-start-to-finish/using-design-patterns/Output/README.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+#### MVC
+MVC is an achitectural pattern that is used to separate an application's concerns. That is, application code is broken into three layers:
+- *Model* - represents an object carrying data. e.g. `InvertedIndex Class`
+- *View* - represents the front end view of the data that model contains. e.g. `index.html`
+- *Controller* - acts on both model and view. It controls the data flow into model object and updates the view whenever data changes. e.g. `app.js`
+
+Hence as explained above, in the case of the Inverted Index App, the model is the `InvertedIndex Class`, the view is the front end HTML and CSS, while the controller is the angular app.
+
+#### Targeted Learning Outcomes
+- [Patterns (MVC)](https://github.com/andela/learningmap/tree/master/Phase-C/Entry-level%20Developer/Curriculum/23%20-%20Patterns%20(MVC))
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/using-design-patterns/Output/README.md:64]: Looks good!

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/README.md [hunk 21]
```diff
@@ -3,6 +3,8 @@
 - **Deliver**: a folder containing the template for the proposed frontend of your project
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/README.md:3]: Can you please include a screenshot of your front-end. Thanks!

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 22]
```diff
@@ -0,0 +1,124 @@
...
+    });
+
+    it('should have an indexed object to hold all indexes', () => {
+      expect(typeof this.invertedIndex.indexed).toEqual('object');
+    });
+  });
+
+  describe('The readFile method', () => {
+    it('should read a file', () => {
+      beforeEach(() => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: Why is this `beforeEach` in the spec rather than the suite?

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+    });
+
+    it('should return the false if the file cannot be parsed', () => {
+      const readFile = InvertedIndex.readFile(alice);
+      expect(typeof readFile).toEqual('object');
+    });
+  });
+
+  describe('createIndex function', () => {
+    it('should create an index for a valid file', () => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js:66]: Let's test the exact value of the index instead of just one word.

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 24]
```diff
@@ -0,0 +1,124 @@
...
+    });
+
+    it('should return the correct array or words', () => {
+      expect(InvertedIndex.tokenize(words))
+        .toEqual(['ralia', 'the', 'funk', 'and', 'now', 'this', 'is', 
+          'just', 'rowland']);
+    });
+  });
+
+  describe('searchIndex function', () => {
```
Comment (q-ode) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js:146]: I'd like to see you test the exact search result. 

Also include tests for searching through one file and tests for searching through all index files.

**Review by q-ode — COMMENTED** [review]
I've got to say, this is brilliant work!

Please attend to the comments in the PR.

👍

File: 06-building-a-program-start-to-finish/automating-tasks/Output/README.md (hunk 1 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/automating-tasks/Output/README.md:7]: Thank you!

File: 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md (hunk 2 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md:12]: Noted

File: 06-building-a-program-start-to-finish/getting-peer-reviews/Output/README.md (hunk 4 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/getting-peer-reviews/Output/README.md:7]: Noted

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md (hunk 5 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md:7]: Thank you

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.md (hunk 6 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/Output/README.md:1]: changed

File: 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md (hunk 10 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md]: Included

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 11 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: done

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 13 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Thank you

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 14 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Updated

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 12 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: done

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 16 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Noted

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 17 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: updated to tokens

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 19 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Updated

File: 06-building-a-program-start-to-finish/using-design-patterns/Output/README.md (hunk 20 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/using-design-patterns/Output/README.md:64]: Thank you

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/README.md (hunk 21 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/README.md:3]: Done

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 22 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: updated

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 23 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js:66]: updated

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 24 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js:146]: updated

File: 06-building-a-program-start-to-finish/setting-up-linters/Output/README.md (hunk 8 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-linters/Output/README.md:1]: Noted

File: 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md (hunk 3 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md:7]: Updated

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md (hunk 7 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md]: My apologies, updated

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md (hunk 9 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md]: Updated

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 15 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Updated

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 18 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Done

File: 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md (hunk 3 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md:7]: Seen.

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md [hunk 25]
```diff
@@ -3,6 +3,12 @@
 - **Deliver**: a link to a successful TravisCI build of your `development` branch (it should include test runs)
 - **With**: *independently*
 
+### Link to a successful Travis build on development branch
+[Travis CI](https://travis-ci.org/andela-tezebuike/inverted-index/builds/218060350)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md:7]: Seen.

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+#### Here are links to the `review`, `staging` & `production` apps
+ - [Review]
+ (https://review-indexx-app.herokuapp.com/)
+
+ - [Staging]
+ (https://invertedindex-staging.herokuapp.com/index.html)
+
+ - [Production]
+ (https://indexx-app.herokuapp.com/index.html)
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md]: Live URLs. However, why are the URL names pattern somewhat inconsistent?

File: 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md (hunk 10 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md]: Seen the new inclusion, however I believe it can be properly formatted Such that it doesn't look disconnect from the other content on the page.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 11 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Please ensure this file's name (**_invertedindex.js_**) is in PascalCase as prescribed by our style guides.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 27]
```diff
@@ -0,0 +1,167 @@
...
+            indices[word].push(index);
+          }
+        } else {
+          indices[word] = [index];
+        }
+      });
+    });
+    this.indexed[fileName] = {
+      eachWord: indices,
+      numOfDocs: books.length
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Using abbreviations for your arguments/parameter/variable names are not allowed. It doesn't aid readability

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 28]
```diff
@@ -0,0 +1,167 @@
...
+  searchIndex(phrase, filename) {
+    let result = {};
+    if (filename === 'All') {
+      const files = this.indexed;
+      Object.keys(files).forEach((file) => {
+        const storedIndex = this.getIndex(file);
+        const searchWords = InvertedIndex.tokenize(phrase);
+        const search = {
+          eachWord: {},
+          numOfDocs: storedIndex.numOfDocs
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: (**_numOfDocs_**) Using abbreviations for your arguments/parameter/variable names are not allowed. It doesn't aid readability

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js [hunk 29]
```diff
@@ -0,0 +1,167 @@
...
+          if (!(search.eachWord[word] in search)) {
+            search.eachWord[word] = storedIndex.eachWord[word];
+          }
+        } else search.eachWord[word] = [];
+      });
+      result = search;
+    }
+    return result;
+  }
+}
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: It seems there are snippets of code which you repeat in your searchIndex(). I believe you can make it a lot more tidy

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 30]
```diff
@@ -0,0 +1,178 @@
+const alice = require('../books/alice.json');
+const rabbits = require('../books/rabbits.json');
+const wrongkeys = require('../books/wrongkeys.json');
+const oneKey = require('../books/oneKey.json');
+const emptyString = require('../books/emptystring.json');
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: A base filename should exactly match the name of its default export. [See link](https://github.com/airbnb/javascript#naming--filename-matches-export)

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 31]
```diff
@@ -0,0 +1,178 @@
...
+describe('InvertedIndex Test Suite', () => {
+  beforeAll(() => {
+    this.invertedIndex = new InvertedIndex();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: Why are you instantiating your class this way? Is it possible to instantiate a variable instead.

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 32]
```diff
@@ -0,0 +1,178 @@
...
+
+    it('should have an indexed object to hold all indexes', () => {
+      expect(typeof this.invertedIndex.indexed).toEqual('object');
+    });
+  });
+
+  describe('The readFile method', () => {
+    it('should read a file', () => {
+      const reader = new FileReader();
+      reader.addEventListener('load', (e) => {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: What is **_e_**? Please be more descriptive with your arguments/variables/parameter names

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 33]
```diff
@@ -0,0 +1,178 @@
...
+    });
+
+    it('should call validateFile', () => {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        const fileRead = JSON.parse(reader.result);
+        expect(InvertedIndex.validateFile(fileRead)).toHaveBeenCalled;
+        done();
+      });
+    });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: I am not completely sure about this test spec. Does it not belong in a different test suite?

**Review by andela-moe — COMMENTED** [review]
Please address comments

File: 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md (hunk 25 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-continuous-integration-(build-server)/README.md:7]: Noted

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md (hunk 26 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md]: Updated

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 27 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: updated to count

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 28 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: updated to count

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 30 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: Updated

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 31 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: instantiated now with a variable

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 32 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: updated. Thanks

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 33 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: Actually, i am testing to ensure it calls the validateFile method from the readFile method.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js (hunk 29 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/invertedindex.js]: Code has been refactored. Thanks!

**Comment by tezebuike** [comment]
Thanks a lot Mark for the review

**Comment by andela-moe** [comment]
- Hope your eslint is properly configured. You have some eslint violation... `trailing spaces` `max-len violations` on your invertedIndex.test.js
- Line 159 & 175 exceeds the maximum line length of 80. (max-len) on your invertedIndex.test.js
- `'event' is defined but never used.` on line 38

Please fix them.

**Comment by tezebuike** [comment]
Resolved

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 34]
```diff
@@ -0,0 +1,180 @@
+const invertedIndex = new InvertedIndex();
+const alice = require('../books/alice.json');
+const rabbits = require('../books/rabbits.json');
+const wrongkeys = require('../books/wrongkeys.json');
+const oneKey = require('../books/oneKey.json');
+const emptystring = require('../books/emptystring.json');
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: I believe we have talked about this. Variable names should be in camelCase

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js [hunk 35]
```diff
@@ -0,0 +1,180 @@
...
+        expect(JSON.parse(fileRead) instanceof Object).toBeTruthy();
+        done();
+      });
+    });
+
+    it('should call validateFile', () => {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        const fileRead = JSON.parse(reader.result);
+        expect(InvertedIndex.validateFile(fileRead)).toHaveBeenCalled;
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: `.toHaveBeenCalled` or `.toHaveBeenCalled()`

**Review by andela-moe — COMMENTED** [review]
Good work. Almost there I believe. Address comments

File: 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/Output/app.js [hunk 36]
```diff
@@ -0,0 +1,71 @@
...
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  document.getElementById('upload')
+    .addEventListener('change', function createFile() {
+      if (this.files[0]) {
+        angular.element(this).scope().createIndex(this.files[0]);
+      }
+    });
+});
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-html-js-css-outputs-(template)/Output/app.js:71]: So when an incorrectly formatted _**json**_ file is uploaded, what happens? I do not think you are handling that. You can see Faith Mustapha for an explanation for this.

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 34 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: Updated

File: 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js (hunk 35 above)
Comment (tezebuike) [reply 06-building-a-program-start-to-finish/writing-tests/Output/invertedIndex.spec.js]: updated. Thank you

**Comment by andela-moe** [comment]
Good work.
